src/plugins/share/public/components/tabs/link/link_content.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiCodeBlock, EuiForm, EuiSpacer, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React, { useCallback, useEffect, useState } from 'react';
+import useMountedState from 'react-use/lib/useMountedState';
+import { format as formatUrl, parse as parseUrl } from 'url';
+import { IShareContext } from '../../context';
+
```

Comment:
Check out https://github.com/elastic/kibana/blob/b4e4e8cddddd66585ebd7ec311cab27f07218811/src/plugins/share/public/components/link_modal.tsx for the Lens link unsaved state work thanks!